scribus/plugins/import/pdf/slaoutput.cpp

```diff
@@ -3896,3 +3926,680 @@ bool SlaOutputDev::checkClip()
+		//moveto deals with setting newlines and segments, all we hve to do is populatit them with the parameters tyhe glyph gives us such as it's width and height and set the glyph index for the newlines and segments
+
+		TextRegionLine* textRegionLine = &textRegionLines.back();
+		if (pass == FrameworkLineTests::NEWLINE || pass == FrameworkLineTests::FIRSTPOINT) {
+			textRegionLine->glyphIndex = glyphs.size() - 1;
+			textRegionLine->baseOrigin = QPointF(textRegioBasenOrigin.x(), newGlyphPoint.y());
+		}
+
+		TextRegionLine *segment = &textRegionLine->segments.back();
+		segment->width = abs(movedGlyphPoint.x() - segment->baseOrigin.x());
+		segment->glyphIndex = glyphs.size() - 1;
+		qreal thisHeight = 0;
+		if (textRegionLines.size() > 1)
+		{
+			thisHeight = abs(newGlyphPoint.y() - textRegionLines[textRegionLines.size() - 2].baseOrigin.y());
+		}
+		else
+		{
+			thisHeight = newGlyph.dx;
+		}
+		segment->maxHeight = thisHeight > segment->maxHeight ? thisHeight : segment->maxHeight;
+		textRegionLine->maxHeight = textRegionLine->maxHeight > thisHeight ? textRegionLine->maxHeight : thisHeight;
+		textRegionLine->width = abs(movedGlyphPoint.x() - textRegionLine->baseOrigin.x());
+
+		maxWidth = textRegionLine->width > maxWidth ? textRegionLine->width : maxWidth;
+		if (textRegionLine->segments.size() == 1)
+		{
+			lineBaseXY = textRegionLine->baseOrigin;
+		}
+		lastXY = movedGlyphPoint;
+	}
+	else
+	{
+		qDebug() << "addGlyphAtPoint failed, this should never happen";
+	}
+	return pass;
+}
+
+void TextRegion::renderToTextFrame(PageItem* textNode, ParagraphStyle& pStyle)
+{
+	// nothing clever, just get all the body text in one lump and update the text frame
+	textNode->setWidthHeight(this->maxWidth, this->maxHeight);
+	QString bodyText = "";
+	for (int glyphIndex = this->textRegionLines.begin()->glyphIndex; glyphIndex <= this->textRegionLines.back().segments.back().glyphIndex; glyphIndex++)
+	{
+		bodyText += glyphs[glyphIndex].code;
+	}
+	textNode->itemText.insertChars(bodyText);
+	textNode->frameTextEnd();
+}
+
+bool TextRegion::isNew()
+{
+	return textRegionLines.empty() ||
+		glyphs.empty();
+}
+
+void SlaOutputDev::setFillAndStrokeForPDF(GfxState* state, PageItem* textNode)
+{
+
+	textNode->ClipEdited = true;
+	textNode->FrameType = 3;
+	textNode->setLineEnd(PLineEnd);
+	textNode->setLineJoin(PLineJoin);
+	textNode->setTextFlowMode(PageItem::TextFlowDisabled);
+
+	int textRenderingMode = state->getRender();
+	// Invisible or only used for clipping
+	if (textRenderingMode == 3)
+	{
+		return;
+	}
+	// Fill text rendering modes. See above
+	if (textRenderingMode == 0 || textRenderingMode == 2 || textRenderingMode == 4 || textRenderingMode == 6)
+	{
+
+		CurrColorFill = getColor(state->getFillColorSpace(), state->getFillColor(), &CurrFillShade);
+		if (textNode->isTextFrame()) { //fill colour sets the background colour for the frame not the fill colour fore  the text			

```

Comment:
depending if the comment concerns the if or the if-else the comment should be on its own line before or after the if

Reply:
ok